WordPress/Classes/ViewRelated/Me/Account Settings/AccountSettingsViewController.swift

```diff
@@ -228,6 +239,75 @@ private class AccountSettingsController: SettingsController {
         }
     }
 
+    private var closeAccountAction: (ImmuTableRow) -> UIAlertController {
+        return { [weak self] _ in
+            return self?.closeAccountAlert ?? UIAlertController()
+        }
+    }
+
+    private var closeAccountAlert: UIAlertController? {
+        guard let value = settings?.username else {
+            return nil
+        }
+
+        let title = NSLocalizedString("Confirm Close Account", comment: "Close Account alert title")
+        let message = NSLocalizedString("\nTo confirm, please re-enter your username before closing.\n\n",
+                                        comment: "Message of Close Account confirmation alert")
+        let destructiveActionTitle = NSLocalizedString("Permanently Close Account",
+                                                       comment: "Close Account confirmation action title")
+
+        return alertHelper.makeAlertWithConfirmation(title: title, message: message, valueToConfirm: value, destructiveActionTitle: destructiveActionTitle, destructiveAction: closeAccount)
+    }
+
+    private func closeAccount() {
+        let status = NSLocalizedString("Closing account…", comment: "Overlay message displayed while closing account")
+        SVProgressHUD.setDefaultMaskType(.black)
+        SVProgressHUD.show(withStatus: status)
+
+        service.closeAccount { [weak self] in
+            guard let self = self else { return }
+            switch $0 {
+            case .success:
+                let status = NSLocalizedString("Account closed", comment: "Overlay message displayed when account successfully closed")
+                SVProgressHUD.showDismissibleSuccess(withStatus: status)
+                AccountHelper.logOutDefaultWordPressComAccount()
+            case .failure(let error):
+                SVProgressHUD.dismiss()
+                DDLogError("Error closing account: \(error.localizedDescription)")
+                let message = NSLocalizedString(self.generateLocalizedMessage(error), comment: "Error message displayed when unable to close user account.")
+                self.showErrorAlert(message: message)
+            }
+        }
+    }
+
+    private func generateLocalizedMessage(_ error: Error) -> String {
+        let userInfo = (error as NSError).userInfo
+        let errorCode = userInfo[WordPressComRestApi.ErrorKeyErrorCode] as? String
+
+        switch errorCode {
+        case "unauthorized":
+            return "You're not authorized to close the account."
```

Comment:
I think you can do something like this instead:

```suggestion
        case "unauthorized":
            return NSLocalizedString("You're not authorized to close the account.", comment: "Error message displayed when unable to close user account due to being unauthorized.")
```